torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -939,15 +939,23 @@ void ProcessGroupNCCL::enableCollectivesTiming() {
   enableTiming_.store(true);
 }
 
-c10::optional<std::thread> ProcessGroupNCCL::tryWriteDebugInfo() {
-  std::lock_guard<std::mutex> lock(writeDebugInfoMutex_);
-  if (writeDebugInfo_) {
```

Comment:
Well this flag is still needed since we don't want the write is triggered multiple times?

Reply:
Well, without the gating of the flag, then inside monitor thread, after sleeping, it will still try to dump debug info. Or you are saying we should use two different file paths in two threads?

Reply:
I mean with the mutex's protection, the write will probably happen one by one, but since the file name is the same, I am afraid that we will overwrite the existing file.